app/Actions/Albums/Tree.php

```diff
@@ -14,16 +16,20 @@
 class Tree
 {
 	private AlbumAuthorisationProvider $albumAuthorisationProvider;
-	private string $sortingCol;
-	private string $sortingOrder;
+	private AlbumSortingCriterion $sorting;
 
+	/**
+	 * @throws InvalidOrderDirectionException
+	 */
 	public function __construct(AlbumAuthorisationProvider $albumAuthorisationProvider)
 	{
 		$this->albumAuthorisationProvider = $albumAuthorisationProvider;
-		$this->sortingCol = Configs::get_value('sorting_Albums_col', 'created_at');
-		$this->sortingOrder = Configs::get_value('sorting_Albums_order', 'ASC');
+		$this->sorting = AlbumSortingCriterion::createDefault();
 	}
 
+	/**
+	 * @throws InternalLycheeException
+	 */
 	public function get(): array
```

Comment:
Not directly related to this PR but inspired by it, I think that this API call should also return smart and tag albums. While they can't be the target of a Move operation or such, they should be included in the front-end GUI in the pull-down list of all the albums that's available after clicking on the album name in the header.

I'm happy to do the necessary coding but I'm wondering if the return type of this call should be kept an array or whether it should be changed to the `TopAlbums` DTO to match `Albums::get`?

Reply:
> I think that this API call should also return smart and tag albums. While they can't be the target of a Move operation or such, they should be included in the front-end GUI in the pull-down list of all the albums that's available after clicking on the album name in the header.

The latter is a good point. However, how and where you remove the smart albums from the tree for a move operation? On the frontend?

> I'm happy to do the necessary coding but I'm wondering if the return type of this call should be kept an array or whether it should be changed to the `TopAlbums` DTO to match Albums::get?

The return code should probably be a proper DTO to improve type-safty. But unless I am not completely mistaken, it cannot be `TopAlbums`, can it? TopAlbums` only includes the first level of the hierarchy while the returned array of this method returns the complete tree with all subalbums unfolded.

Reply:
I added a proper DTO for the sake of consistency. See this commit: https://github.com/LycheeOrg/Lychee/pull/1217/commits/367ccfc0eb1afb1b5ecd2482853f8007806245d9. However, I did not include smart and tag albums into the tree. So it is still the the old behavior, but cleaned up.

While I was on it, I also create a DTO for `PositionData`. See this commit: https://github.com/LycheeOrg/Lychee/pull/1217/commits/d5c7d2ff12d6365ca574dc2555120cf913396b36.